### ISSUE TITLE
Implement a size overflow check in a base handler class.

### ISSUE THREAD

**Issue by QKaiser** [IoT-Inspector/unblob#22]
All format handlers calculate the expected size of a chunk. We should implement a generic check that verifies if the calculated `end_offset` points after the actual end of a file.

Something along those lines:

```python
def overflow(file: io.BufferedReader, end_offset: int):
    file.seek(0, os.SEEK_END)
    size = file.tell()
    return end_offset > size
```

This is an easy win in terms of heuristics, to get rid of invalid and corrupted headers.

**Comment by kissgyorgy** [comment]
Depends on https://github.com/IoT-Inspector/unblob/issues/21

**Comment by kissgyorgy** [comment]
We should emit an error log about this.

**Comment by QKaiser** [comment]
I fully agree.

**Comment by kissgyorgy** [comment]
Done in #47
